### DISCUSSION TITLE
custom code folder structure in understrap

### DISCUSSION THREAD

**Discussion by joaquiboy** [holger1411/understrap#1403] [Q&A] [answered]
I just want to ask if there are documentations on how we can folder structure our custom codes base on understrap recommended practice in child theme. if you're getting confused about what I mean, it's just we want to break the files and not relying on a single file to put the entire code for all modifications, then we want to hook that certain PHP file into a single file template to make it more organized.

**Comment by bacoords** [comment, marked as answer]
Hi, 

You can add your own functions files in the `/inc/` folder (you'll need to add this folder to your child theme). Then in your functions.php, simply add `require_once get_theme_file_path( inc/my-custom-file.php);` for each file.

Or you can automatically include all of the files in your `/inc/`directory with a loop like this:

```php
	// Include php files from inc folder.
	foreach ( glob( dirname( __FILE__ ) . '/inc/*.php' ) as $file ) {
		include $file;
	}
```

I also recommend prefixing all of your custom functions files (i.e. if your prefix was "abc", you could use `abc-nav.php`, `abc-widget.php`, etc) so that they don't override or conflict with functions files in the parent theme.
